### ISSUE TITLE
Getting Error from Section build

### ISSUE THREAD

**Issue by jbruett** [iainbrighton/PScribo#29]
I'm receiving the following set of errors, though the doc still generates correctly as far as I can tell.

```powershell
The property 'Type' cannot be found on this object. Verify that the property exists.
At C:\Program Files\WindowsPowerShell\Modules\PScribo\0.7.12.47\Src\Private\Document.Internal.ps1:91 char:2
+ ...              if ($s.Type -like '*.Section' -and -not $s.IsExcluded) {
+                      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [], PropertyNotFoundException
    + FullyQualifiedErrorId : PropertyNotFoundStrict

The property 'Type' cannot be found on this object. Verify that the property exists.
At C:\Program Files\WindowsPowerShell\Modules\PScribo\0.7.12.47\Src\Private\Document.Internal.ps1:91 char:2
+ ...              if ($s.Type -like '*.Section' -and -not $s.IsExcluded) {
+                      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [], PropertyNotFoundException
    + FullyQualifiedErrorId : PropertyNotFoundStrict

The property 'Type' cannot be found on this object. Verify that the property exists.
At C:\Program Files\WindowsPowerShell\Modules\PScribo\0.7.12.47\Src\Private\Document.Internal.ps1:91 char:2
+ ...              if ($s.Type -like '*.Section' -and -not $s.IsExcluded) {
+                      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [], PropertyNotFoundException
    + FullyQualifiedErrorId : PropertyNotFoundStrict

The property 'Type' cannot be found on this object. Verify that the property exists.
At C:\Program Files\WindowsPowerShell\Modules\PScribo\0.7.12.47\Src\Private\Document.Internal.ps1:91 char:2
+ ...              if ($s.Type -like '*.Section' -and -not $s.IsExcluded) {
+                      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [], PropertyNotFoundException
    + FullyQualifiedErrorId : PropertyNotFoundStrict

The property 'Type' cannot be found on this object. Verify that the property exists.
At C:\Program Files\WindowsPowerShell\Modules\PScribo\0.7.12.47\Src\Private\Document.Internal.ps1:91 char:2
+ ...              if ($s.Type -like '*.Section' -and -not $s.IsExcluded) {
+                      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [], PropertyNotFoundException
    + FullyQualifiedErrorId : PropertyNotFoundStrict

The property 'Type' cannot be found on this object. Verify that the property exists.
At C:\Program Files\WindowsPowerShell\Modules\PScribo\0.7.12.47\Src\Private\Document.Internal.ps1:91 char:2
+ ...              if ($s.Type -like '*.Section' -and -not $s.IsExcluded) {
+                      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [], PropertyNotFoundException
    + FullyQualifiedErrorId : PropertyNotFoundStrict

The property 'Type' cannot be found on this object. Verify that the property exists.
At C:\Program Files\WindowsPowerShell\Modules\PScribo\0.7.12.47\Src\Private\Document.Internal.ps1:91 char:2
+ ...              if ($s.Type -like '*.Section' -and -not $s.IsExcluded) {
+                      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [], PropertyNotFoundException
    + FullyQualifiedErrorId : PropertyNotFoundStrict

The property 'Type' cannot be found on this object. Verify that the property exists.
At C:\Program Files\WindowsPowerShell\Modules\PScribo\0.7.12.47\Src\Private\Document.Internal.ps1:91 char:2
+ ...              if ($s.Type -like '*.Section' -and -not $s.IsExcluded) {
+                      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [], PropertyNotFoundException
    + FullyQualifiedErrorId : PropertyNotFoundStrict

The property 'Type' cannot be found on this object. Verify that the property exists.
At C:\Program Files\WindowsPowerShell\Modules\PScribo\0.7.12.47\Src\Private\Document.Internal.ps1:91 char:2
+ ...              if ($s.Type -like '*.Section' -and -not $s.IsExcluded) {
+                      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [], PropertyNotFoundException
    + FullyQualifiedErrorId : PropertyNotFoundStrict

The property 'Type' cannot be found on this object. Verify that the property exists.
At C:\Program Files\WindowsPowerShell\Modules\PScribo\0.7.12.47\Src\Private\Document.Internal.ps1:91 char:2
+ ...              if ($s.Type -like '*.Section' -and -not $s.IsExcluded) {
+                      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [], PropertyNotFoundException
    + FullyQualifiedErrorId : PropertyNotFoundStrict

The property 'Id' cannot be found on this object. Verify that the property exists.
At C:\Program Files\WindowsPowerShell\Modules\PScribo\0.7.12.47\Src\Plugins\OutWord.Internal.ps1:66 char:25
+                     if ($s.Id.Length -gt 40) { $sectionId = '{0}[..]' ...
+                         ~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [], PropertyNotFoundException
    + FullyQualifiedErrorId : PropertyNotFoundStrict

The property 'Type' cannot be found on this object. Verify that the property exists.
At C:\Program Files\WindowsPowerShell\Modules\PScribo\0.7.12.47\Src\Plugins\OutWord.Internal.ps1:70 char:21
+ ...             WriteLog -Message ($localized.PluginProcessingSection -f  ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [], PropertyNotFoundException
    + FullyQualifiedErrorId : PropertyNotFoundStrict

The property 'Type' cannot be found on this object. Verify that the property exists.
At C:\Program Files\WindowsPowerShell\Modules\PScribo\0.7.12.47\Src\Plugins\OutWord.Internal.ps1:70 char:21
+ ...             WriteLog -Message ($localized.PluginProcessingSection -f  ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [], PropertyNotFoundException
    + FullyQualifiedErrorId : PropertyNotFoundStrict

The property 'Id' cannot be found on this object. Verify that the property exists.
At C:\Program Files\WindowsPowerShell\Modules\PScribo\0.7.12.47\Src\Plugins\OutWord.Internal.ps1:66 char:25
+                     if ($s.Id.Length -gt 40) { $sectionId = '{0}[..]' ...
+                         ~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [], PropertyNotFoundException
    + FullyQualifiedErrorId : PropertyNotFoundStrict

The property 'Type' cannot be found on this object. Verify that the property exists.
At C:\Program Files\WindowsPowerShell\Modules\PScribo\0.7.12.47\Src\Plugins\OutWord.Internal.ps1:70 char:21
+ ...             WriteLog -Message ($localized.PluginProcessingSection -f  ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [], PropertyNotFoundException
    + FullyQualifiedErrorId : PropertyNotFoundStrict

The property 'Type' cannot be found on this object. Verify that the property exists.
At C:\Program Files\WindowsPowerShell\Modules\PScribo\0.7.12.47\Src\Plugins\OutWord.Internal.ps1:70 char:21
+ ...             WriteLog -Message ($localized.PluginProcessingSection -f  ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [], PropertyNotFoundException
    + FullyQualifiedErrorId : PropertyNotFoundStrict

The property 'Id' cannot be found on this object. Verify that the property exists.
At C:\Program Files\WindowsPowerShell\Modules\PScribo\0.7.12.47\Src\Plugins\OutWord.Internal.ps1:66 char:25
+                     if ($s.Id.Length -gt 40) { $sectionId = '{0}[..]' ...
+                         ~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [], PropertyNotFoundException
    + FullyQualifiedErrorId : PropertyNotFoundStrict

The property 'Type' cannot be found on this object. Verify that the property exists.
At C:\Program Files\WindowsPowerShell\Modules\PScribo\0.7.12.47\Src\Plugins\OutWord.Internal.ps1:70 char:21
+ ...             WriteLog -Message ($localized.PluginProcessingSection -f  ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [], PropertyNotFoundException
    + FullyQualifiedErrorId : PropertyNotFoundStrict

The property 'Type' cannot be found on this object. Verify that the property exists.
At C:\Program Files\WindowsPowerShell\Modules\PScribo\0.7.12.47\Src\Plugins\OutWord.Internal.ps1:70 char:21
+ ...             WriteLog -Message ($localized.PluginProcessingSection -f  ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [], PropertyNotFoundException
    + FullyQualifiedErrorId : PropertyNotFoundStrict

The property 'Id' cannot be found on this object. Verify that the property exists.
At C:\Program Files\WindowsPowerShell\Modules\PScribo\0.7.12.47\Src\Plugins\OutWord.Internal.ps1:66 char:25
+                     if ($s.Id.Length -gt 40) { $sectionId = '{0}[..]' ...
+                         ~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [], PropertyNotFoundException
    + FullyQualifiedErrorId : PropertyNotFoundStrict

The property 'Type' cannot be found on this object. Verify that the property exists.
At C:\Program Files\WindowsPowerShell\Modules\PScribo\0.7.12.47\Src\Plugins\OutWord.Internal.ps1:70 char:21
+ ...             WriteLog -Message ($localized.PluginProcessingSection -f  ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [], PropertyNotFoundException
    + FullyQualifiedErrorId : PropertyNotFoundStrict

The property 'Type' cannot be found on this object. Verify that the property exists.
At C:\Program Files\WindowsPowerShell\Modules\PScribo\0.7.12.47\Src\Plugins\OutWord.Internal.ps1:70 char:21
+ ...             WriteLog -Message ($localized.PluginProcessingSection -f  ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [], PropertyNotFoundException
    + FullyQualifiedErrorId : PropertyNotFoundStrict

The property 'Id' cannot be found on this object. Verify that the property exists.
At C:\Program Files\WindowsPowerShell\Modules\PScribo\0.7.12.47\Src\Plugins\OutWord.Internal.ps1:66 char:25
+                     if ($s.Id.Length -gt 40) { $sectionId = '{0}[..]' ...
+                         ~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [], PropertyNotFoundException
    + FullyQualifiedErrorId : PropertyNotFoundStrict

The property 'Type' cannot be found on this object. Verify that the property exists.
At C:\Program Files\WindowsPowerShell\Modules\PScribo\0.7.12.47\Src\Plugins\OutWord.Internal.ps1:70 char:21
+ ...             WriteLog -Message ($localized.PluginProcessingSection -f  ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [], PropertyNotFoundException
    + FullyQualifiedErrorId : PropertyNotFoundStrict

The property 'Type' cannot be found on this object. Verify that the property exists.
At C:\Program Files\WindowsPowerShell\Modules\PScribo\0.7.12.47\Src\Plugins\OutWord.Internal.ps1:70 char:21
+ ...             WriteLog -Message ($localized.PluginProcessingSection -f  ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [], PropertyNotFoundException
    + FullyQualifiedErrorId : PropertyNotFoundStrict

The property 'Id' cannot be found on this object. Verify that the property exists.
At C:\Program Files\WindowsPowerShell\Modules\PScribo\0.7.12.47\Src\Plugins\OutWord.Internal.ps1:66 char:25
+                     if ($s.Id.Length -gt 40) { $sectionId = '{0}[..]' ...
+                         ~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [], PropertyNotFoundException
    + FullyQualifiedErrorId : PropertyNotFoundStrict

The property 'Type' cannot be found on this object. Verify that the property exists.
At C:\Program Files\WindowsPowerShell\Modules\PScribo\0.7.12.47\Src\Plugins\OutWord.Internal.ps1:70 char:21
+ ...             WriteLog -Message ($localized.PluginProcessingSection -f  ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [], PropertyNotFoundException
    + FullyQualifiedErrorId : PropertyNotFoundStrict

The property 'Type' cannot be found on this object. Verify that the property exists.
At C:\Program Files\WindowsPowerShell\Modules\PScribo\0.7.12.47\Src\Plugins\OutWord.Internal.ps1:70 char:21
+ ...             WriteLog -Message ($localized.PluginProcessingSection -f  ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [], PropertyNotFoundException
    + FullyQualifiedErrorId : PropertyNotFoundStrict

The property 'Id' cannot be found on this object. Verify that the property exists.
At C:\Program Files\WindowsPowerShell\Modules\PScribo\0.7.12.47\Src\Plugins\OutWord.Internal.ps1:66 char:25
+                     if ($s.Id.Length -gt 40) { $sectionId = '{0}[..]' ...
+                         ~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [], PropertyNotFoundException
    + FullyQualifiedErrorId : PropertyNotFoundStrict

The property 'Type' cannot be found on this object. Verify that the property exists.
At C:\Program Files\WindowsPowerShell\Modules\PScribo\0.7.12.47\Src\Plugins\OutWord.Internal.ps1:70 char:21
+ ...             WriteLog -Message ($localized.PluginProcessingSection -f  ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [], PropertyNotFoundException
    + FullyQualifiedErrorId : PropertyNotFoundStrict

The property 'Type' cannot be found on this object. Verify that the property exists.
At C:\Program Files\WindowsPowerShell\Modules\PScribo\0.7.12.47\Src\Plugins\OutWord.Internal.ps1:70 char:21
+ ...             WriteLog -Message ($localized.PluginProcessingSection -f  ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [], PropertyNotFoundException
    + FullyQualifiedErrorId : PropertyNotFoundStrict

The property 'Id' cannot be found on this object. Verify that the property exists.
At C:\Program Files\WindowsPowerShell\Modules\PScribo\0.7.12.47\Src\Plugins\OutWord.Internal.ps1:66 char:25
+                     if ($s.Id.Length -gt 40) { $sectionId = '{0}[..]' ...
+                         ~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [], PropertyNotFoundException
    + FullyQualifiedErrorId : PropertyNotFoundStrict

The property 'Type' cannot be found on this object. Verify that the property exists.
At C:\Program Files\WindowsPowerShell\Modules\PScribo\0.7.12.47\Src\Plugins\OutWord.Internal.ps1:70 char:21
+ ...             WriteLog -Message ($localized.PluginProcessingSection -f  ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [], PropertyNotFoundException
    + FullyQualifiedErrorId : PropertyNotFoundStrict

The property 'Type' cannot be found on this object. Verify that the property exists.
At C:\Program Files\WindowsPowerShell\Modules\PScribo\0.7.12.47\Src\Plugins\OutWord.Internal.ps1:70 char:21
+ ...             WriteLog -Message ($localized.PluginProcessingSection -f  ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [], PropertyNotFoundException
    + FullyQualifiedErrorId : PropertyNotFoundStrict

The property 'Id' cannot be found on this object. Verify that the property exists.
At C:\Program Files\WindowsPowerShell\Modules\PScribo\0.7.12.47\Src\Plugins\OutWord.Internal.ps1:66 char:25
+                     if ($s.Id.Length -gt 40) { $sectionId = '{0}[..]' ...
+                         ~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [], PropertyNotFoundException
    + FullyQualifiedErrorId : PropertyNotFoundStrict

The property 'Type' cannot be found on this object. Verify that the property exists.
At C:\Program Files\WindowsPowerShell\Modules\PScribo\0.7.12.47\Src\Plugins\OutWord.Internal.ps1:70 char:21
+ ...             WriteLog -Message ($localized.PluginProcessingSection -f  ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [], PropertyNotFoundException
    + FullyQualifiedErrorId : PropertyNotFoundStrict

The property 'Type' cannot be found on this object. Verify that the property exists.
At C:\Program Files\WindowsPowerShell\Modules\PScribo\0.7.12.47\Src\Plugins\OutWord.Internal.ps1:70 char:21
+ ...             WriteLog -Message ($localized.PluginProcessingSection -f  ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [], PropertyNotFoundException
    + FullyQualifiedErrorId : PropertyNotFoundStrict

The property 'Id' cannot be found on this object. Verify that the property exists.
At C:\Program Files\WindowsPowerShell\Modules\PScribo\0.7.12.47\Src\Plugins\OutWord.Internal.ps1:66 char:25
+                     if ($s.Id.Length -gt 40) { $sectionId = '{0}[..]' ...
+                         ~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [], PropertyNotFoundException
    + FullyQualifiedErrorId : PropertyNotFoundStrict

The property 'Type' cannot be found on this object. Verify that the property exists.
At C:\Program Files\WindowsPowerShell\Modules\PScribo\0.7.12.47\Src\Plugins\OutWord.Internal.ps1:70 char:21
+ ...             WriteLog -Message ($localized.PluginProcessingSection -f  ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [], PropertyNotFoundException
    + FullyQualifiedErrorId : PropertyNotFoundStrict

The property 'Type' cannot be found on this object. Verify that the property exists.
At C:\Program Files\WindowsPowerShell\Modules\PScribo\0.7.12.47\Src\Plugins\OutWord.Internal.ps1:70 char:21
+ ...             WriteLog -Message ($localized.PluginProcessingSection -f  ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [], PropertyNotFoundException
    + FullyQualifiedErrorId : PropertyNotFoundStrict
```

From this specific block of the larger block listed below.  If I comment this code out it generates as expected (minus the section that's causing the problem), without errors.

```powershell
            Section -Name "A VPC Subnet Configuration" -Style Heading3 {
                $VpcSubnets = $Subnets | ?{$_.vpcid -eq $vpc.vpcid}
                [System.Collections.ArrayList]$SubnetRows = @()

                foreach($VpcSubnet in $VpcSubnets) {
                    $SubnetRow = new-object psobject | Select-Object Name,CidrBlock,SubnetID,AvailabilityZone,RouteTable
                    $RouteTable = Get-EC2RouteTable -Filter @{name='association.subnet-id';value=$vpcsubnet.subnetid}

                    if($VpcSubnet.tags.count -gt 0){
                        $NameFound = $false
                        foreach($tag in $VpcSubnet.Tags){
                            if(-not $NameFound){
                                $VpcSubnetName = [string]::empty
                                switch ($tag.key){
                                    'Name' {
                                        $VpcSubnetName = $tag.value
                                        $NameFound = $true
                                    }
                                    default {
                                        if($vpc.isdefault){
                                            $VpcSubnetName = "{0}" -f (get-culture).textinfo.toTitleCase("default subnet")
                                        }else{
                                            $VpcSubnetName = "{0}" -f (get-culture).textinfo.toupper("untagged")
                                        }
                                    }
                                }
                            }
                        }

                    }else{
                        $VpcSubnetName = "{0}" -f (get-culture).textinfo.toupper("untagged")
                    }

                    $SubnetRow.Name = $VPCSubnetName
                    $SubnetRow.CidrBlock = $vpcsubnet.CidrBlock
                    $SubnetRow.SubnetID = $VpcSubnet.SubnetID
                    $SubnetRow.AvailabilityZone = $VpcSubnet.AvailabilityZone
                    if($RouteTable -ne $null){
                        $SubnetRow.RouteTable = $RouteTable.RouteTableId
                    } else {
                        $SubnetRow.RouteTable = 'None'
                    }

                    $SubnetRows.add($SubnetRow)
                }

                $SubnetRows | Table -Columns Name,CidrBlock,SubnetID,AvailabilityZone,RouteTable -Headers 'Name','Cidr Block','Subnet ID','Availability Zone','Route Table'
            }
```

```powershell
foreach($region in $customerRegions) {
    Section -Name ("Subnet Configuration ({0})" -f (get-culture).textinfo.ToUpper($region.tolower())) -Style Heading2 {
        foreach($vpc in $Vpcs){
            $VpcName = [string]::empty
            if($vpc.tags.count -gt 0){
                foreach($tag in $vpc.Tags){
                    if($tag.key -eq 'Name' -and ($tag.value -ne [string]::empty -or $tag.value -ne '')) {
                        $VpcName = $tag.value
                    }elseif($vpc.isdefault){
                        $VpcName = "{0}" -f (get-culture).textinfo.toTitleCase("default")
                    }else{
                        $VpcName = "{0}" -f (get-culture).textinfo.toupper("untagged vpc")
                    }
                }
            }else{
                $VpcName = "{0} VPC Subnet Configuration" -f (get-culture).textinfo.toupper("untagged vpc")
            }
            Section -Name "A VPC Subnet Configuration" -Style Heading3 {
                $VpcSubnets = $Subnets | ?{$_.vpcid -eq $vpc.vpcid}
                [System.Collections.ArrayList]$SubnetRows = @()

                foreach($VpcSubnet in $VpcSubnets) {
                    $SubnetRow = new-object psobject | Select-Object Name,CidrBlock,SubnetID,AvailabilityZone,RouteTable
                    $RouteTable = Get-EC2RouteTable -Filter @{name='association.subnet-id';value=$vpcsubnet.subnetid}

                    if($VpcSubnet.tags.count -gt 0){
                        $NameFound = $false
                        foreach($tag in $VpcSubnet.Tags){
                            if(-not $NameFound){
                                $VpcSubnetName = [string]::empty
                                switch ($tag.key){
                                    'Name' {
                                        $VpcSubnetName = $tag.value
                                        $NameFound = $true
                                    }
                                    default {
                                        if($vpc.isdefault){
                                            $VpcSubnetName = "{0}" -f (get-culture).textinfo.toTitleCase("default subnet")
                                        }else{
                                            $VpcSubnetName = "{0}" -f (get-culture).textinfo.toupper("untagged")
                                        }
                                    }
                                }
                            }
                        }

                    }else{
                        $VpcSubnetName = "{0}" -f (get-culture).textinfo.toupper("untagged")
                    }

                    $SubnetRow.Name = $VPCSubnetName
                    $SubnetRow.CidrBlock = $vpcsubnet.CidrBlock
                    $SubnetRow.SubnetID = $VpcSubnet.SubnetID
                    $SubnetRow.AvailabilityZone = $VpcSubnet.AvailabilityZone
                    if($RouteTable -ne $null){
                        $SubnetRow.RouteTable = $RouteTable.RouteTableId
                    } else {
                        $SubnetRow.RouteTable = 'None'
                    }

                    $SubnetRows.add($SubnetRow)
                }

                $SubnetRows | Table -Columns Name,CidrBlock,SubnetID,AvailabilityZone,RouteTable -Headers 'Name','Cidr Block','Subnet ID','Availability Zone','Route Table'
            }
        }
    }
}
```

**Comment by iainbrighton** [comment]
Hi @jbruett - It looks like you've somehow ended up with a $null section? Can you do me a favour and try exporting to Html (or Text)  with `Export-Document -Format Html` and see if you get the same error(s)? I want to see whether it's an issue with the Word plugin or your document... Thanks!

**Comment by jbruett** [comment]
Hey Ian, thansk for taking a look, the docs create perfectly in both situations, but both give similar errors.  I exported to HTML and the errors above that show in the OutWord script file are showing as OutHTML instead.

**Comment by jbruett** [comment]
So I've done a bit more digging and it looks to me like the issue may be centered around scoping and variable availability.  It seems to me that each of the Sections becomes its own scope, therefor wiping out the variables outside of that section.  This wasn't what I expected.  Can you confirm that's expected behavior?

**Comment by iainbrighton** [comment]
Hey @jbruett - the PScribo sections are nested as they can be x levels deep. The plugin export functions do need to recursively enumerate the sections. I think you're somehow ending up with an empty section, but this is in the PScribo document creation, not the export. Is there anything that can't be shared publicly in the script that's not working?

**Comment by jbruett** [comment]
@iainbrighton unfortunately i can't really share it directly for company data protection reasons.  Here's an example of what i think I'm seeing

```Powershell
Section -Name 'Section 1' {
      $x = 3
      $z = 'some thing'
      foreach($a in $x) {
          Section -Name "Section 1.$a" {
                if ($z -eq 'some thing'){                     #this fails, $z is an unknown variable
                   # do a thing
                }
          }
      }
}
```

**Comment by iainbrighton** [comment]
@jbruett No problem - we'll get there in the end!

Are you missing any parts/sections in the final Word document? If you don't `Export-Document`, do you receive any errors when running the `Document { }` scriptblock?

The example given should work as the `$z` will be found in the parent scope due to dynamic scoping:
```powershell
[2] C:\..\Dropbox\PowerShell> 
Document 'Issue29' {
    Section -Name 'Section 1' {
          $x = 3
          $z = 'some thing'
          foreach($a in $x) {
              Section -Name "Section 1.$a" {
                    if ($z -eq 'some thing'){                     #this fails, $z is an unknown variable
                       Paragraph 'Doing a thing' # do a thing
                    }
              }
          }
    }
} | Export-Document -Format Text -Path C:\Users\Iain\Desktop | Get-Content


Section 1
-------------------------------------------------------------------------------------------------------------------

Section 1.3
-------------------------------------------------------------------------------------------------------------------
Doing a thing

```

**Comment by iainbrighton** [comment]
Hi @jbruett - I've found your issue. It's caused by this line `$SubnetRows.add($SubnetRow)` passing data back down the pipeline (and that doesn't have an `Id` or `Type` property like it's expecting). If you change it to `$null = $SubnetRows.add($SubnetRow)` you should be good to go 😄!

**Comment by jbruett** [comment]
I'll have a look and post the outcome later today.  Thanks Iain (sorry for misspelling earlier ☹️)

**Comment by iainbrighton** [comment]
@jbruett I've published v0.7.13.51 that should provide some warnings (instead of errors) for this in future:
```powershell
WARNING: [ 21:55:15:051 ] [ Document ] - Unexpected object 'System.Int32' in section 'A VPC Subnet Configuration'.
WARNING: [ 21:55:15:056 ] [ Document ] - Unexpected object 'System.Int32' in section 'A VPC Subnet Configuration'.
WARNING: [ 21:55:15:059 ] [ Document ] - Unexpected object 'System.Int32' in section 'A VPC Subnet Configuration'.
WARNING: [ 21:55:15:064 ] [ Document ] - Unexpected object 'System.Int32' in section 'A VPC Subnet Configuration'.

Subnet Configuration (EU-WEST-1)
------------------------------------------------------------------------------------------------------------------------

A VPC Subnet Configuration
------------------------------------------------------------------------------------------------------------------------

Name         Cidr Block        Subnet ID       Availability Zone Route Table 
----         ----------        ---------       ----------------- ----------- 
EU-West-XXX1 10.20.254.160/27 subnet-abcd1234 eu-west-1b        rtb-12345678
EU-West-YYY1 10.20.254.224/27 subnet-abcd1235 eu-west-1b        rtb-12345678
EU-West-YYY2 10.20.254.192/27 subnet-abcd1236 eu-west-1a        rtb-12345678
EU-West-XXX2 10.20.254.128/27 subnet-abcd1237 eu-west-1a        rtb-12345678

```

**Comment by jbruett** [comment]
So Iain, I updated my version.  Ran it, got the warnings, put in the fix above and the warnings disappeared.  However, the larger issue of variable scoping is still active (I didn't expect your fix to change that).  I've been stepping through my script execution (via VS Code) and it seems to me that when you move into a subsection all automatic variables are cleared.  I know you mentioned your expectation is that parent scope should persist, that's just not what I'm seeing.  I'll do some additional testing to see if I can reproduce outside of Code.
